src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -523,7 +523,7 @@ public bool TryGetValue<TValue>(SyntaxTree tree, SyntaxTreeValueProvider<TValue>
         private readonly Compilation _compilation;
         private readonly AnalyzerOptions _options;
         private readonly Action<Diagnostic> _reportDiagnostic;
-        private readonly Func<Diagnostic, bool> _isSupportedDiagnostic;
+        private readonly Func<Diagnostic, CancellationToken, bool> _isSupportedDiagnostic;
         private readonly CompilationAnalysisValueProviderFactory? _compilationAnalysisValueProviderFactoryOpt;
         private readonly CancellationToken _cancellationToken;
```

Comment:
>private readonly CancellationToken _cancellationToken; [](http://example.com/codeflow?start=8&length=54)

Is it fine to have this field?

Reply:
This is exposed as a public property on every analysis context type, so yes this is intentional